packages/flutter/lib/src/widgets/tap_and_drag_gestures.dart

```diff
@@ -689,9 +684,9 @@ mixin _TapStatusTrackerMixin on OneSequenceGestureRecognizer {
 ///
 /// When competing against [DragGestureRecognizer], if the pointer does not move a sufficient
 /// global distance to be considered a drag, the recognizers will tie in the arena. If the
-/// pointer does travel enough distance then the [TapAndDragGestureRecognizer] will lose because
-/// the [DragGestureRecognizer] will declare self-victory when the drag threshold is met.
-class TapAndDragGestureRecognizer extends OneSequenceGestureRecognizer with _TapStatusTrackerMixin {
+/// pointer does travel enough distance then the recognizer that entered the arena
+/// first will win. The gesture detected in this case is a drag.
+abstract class TapAndDragGestureRecognizer extends OneSequenceGestureRecognizer with _TapStatusTrackerMixin {
```

Comment:
This could be a breaking change.

Did you think about keeping this as a non-abstract class and not having TapAndPanGestureRecognizer, or is that not possible?  I'm not saying you should do that instead, just wondering about the tradeoffs.

Reply:
I may have missed something, why not just have a TapAndDragGestureRecongnizerBase to be used by TapAndDragGestureRecongnizer and TapAndVerticalDragGestureRecongnizer
TapAndVHorizontalDragGestureRecongnizer